packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -713,7 +724,7 @@ class _ExpansionTileState extends State<ExpansionTile> with SingleTickerProvider
                 leading: widget.leading ?? _buildLeadingIcon(context),
                 title: widget.title,
                 subtitle: widget.subtitle,
-                trailing: widget.trailing ?? _buildTrailingIcon(context),
+                trailing: trailing,
```

Comment:
To me it makes more sense to have the `trailing` widget be not shown even if you provide it if `showTrailingIcon` is false:

```suggestion
                trailing: widget.showTrailingIcon ? widget.trailing ?? _buildTrailingIcon(context) : null,
```

Reply:
The assertion in the constructor prevents this scenario. If `widget.showTrailingIcon` is `false` then `widget.trailing` must be `null`. If I were to change it, it would probably need a new name `showTrailingWidget` or something. I'll wait for someone else's input on this.

Reply:
I think it could keep the name.  Mainly, I'd be surprised if I set the `trailing` icon, and then used the bool to turn its visibility on and off, only to find it would throw an assertion whenever it was false. I think more intuitive would to be allowed to set or not set the `trailing` icon, and the bool strictly says whether one is shown, and if `trailing` is null, then the default icon gets shown when `showTrailingIcon` is true.